src/base_object-inl.h

```diff
@@ -32,10 +32,21 @@
 
 namespace node {
 
+namespace {
+// This just has to be different from the Chromium ones:
+// https://source.chromium.org/chromium/chromium/src/+/main:gin/public/gin_embedders.h;l=18-23;drc=5a758a97032f0b656c3c36a3497560762495501a
+// Otherwise, when Node is loaded in an isolate which uses cppgc, cppgc will
+// misinterpret the data stored in the embedder fields and try to garbage
+// collect them.
+uint16_t kNodeEmbedderId = 0x90de;
+}
+
 BaseObject::BaseObject(Environment* env, v8::Local<v8::Object> object)
     : persistent_handle_(env->isolate(), object), env_(env) {
   CHECK_EQ(false, object.IsEmpty());
-  CHECK_GT(object->InternalFieldCount(), 0);
+  CHECK_GT(object->InternalFieldCount(), BaseObject::kSlot);
```

Comment:
IIUC, the internal field count should now be >= 2 because now it's kEmbedderType and kSlot, right?
```suggestion
  CHECK_GE(object->InternalFieldCount(), BaseObject::kInternalFieldCount);
```

Reply:
Right, but the updated check now says it simply needs to be greater than 1, so is your change not semantically identical? (> 1 vs >= 2)

Reply:
Yes, both of these are identical but kInternalFieldCount is used to indicate a count which is not an actual used as an internal field index unlike things like kSlot and kEmbedderType which are valid internal field indices